mmv1/products/firebasehosting/CustomDomain.yaml

```diff
@@ -0,0 +1,514 @@
+    "{{project}}/{{site_id}}/{{custom_domain}}",
+    "sites/{{site_id}}/customDomains/{{custom_domain}}",
+    "{{site_id}}/{{custom_domain}}",
+  ]
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  actions: ["update", "delete"]
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: "name"
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: "response"
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: "done"
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: "error"
+    message: "message"
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  post_create: templates/terraform/post_create/firebasehosting_custom_domain_wait_dns_verification.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebasehosting_customdomain_basic"
+    min_version: "beta"
+    primary_resource_id: "default"
+    vars:
+      site_id: site-id
+      custom_domain: custom.domain.com
+    test_env_vars:
+      project_id: :PROJECT_NAME
+    test_vars_overrides:
+      # Use the default hosting site for the project in tests
+      site_id: acctest.GetTestProjectFromEnv()
+      # Don't add random suffix
+      custom_domain: '"custom.domain.com"'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebasehosting_customdomain_full"
+    min_version: "beta"
+    primary_resource_id: "default"
+    vars:
+      site_id: site-id
+      custom_domain: source.domain.com
+      redirect_target: destination.domain.com
+    test_env_vars:
+      project_id: :PROJECT_NAME
+    test_vars_overrides:
+      # Use the default hosting site for the project in tests
+      site_id: acctest.GetTestProjectFromEnv()
+      # Don't add random suffix
+      custom_domain: '"source.domain.com"'
+      redirect_target: '"destination.domain.com"'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebasehosting_customdomain_cloud_run"
+    min_version: "beta"
+    primary_resource_id: "default"
+    vars:
+      site_id: site-id
+      custom_domain: run.custom.domain.com
+      cloud_run_service_id: cloud-run-service-via-hosting
+    test_env_vars:
+      project_id: :PROJECT_NAME
+    test_vars_overrides:
+      # Don't add random suffix
+      custom_domain: '"run.custom.domain.com"'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: site_id
+    description: |
+      The ID of the site in which to create this custom domain association.
+    immutable: true
+    required: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: custom_domain
+    description: |
+      The ID of the `CustomDomain`, which is the domain name you'd like to use with Firebase Hosting.
+    immutable: true
+    required: true
+    url_param_only: true
+  - !ruby/object:Api::Type::Boolean
+    name: wait_dns_verification
```

Comment:
Could you clarify why this is a parameter instead of a property?

Reply:
This field is not part of the CustomDomain entity. It is used to control the creation behavior of Terraform whether to wait for DNS verification before considering this resource as created. IIUC, this is a use case for a parameter, right?

Reply:
Had to look it up myself, and came across [`virtual_fields`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/virtual_fields.rb). I think you may want to try that instead, as it mentions you can get similar behavior with `url_param_only`, but semantically it seems like `virtual_fields` is a better match for what you are doing.

If for some reason that won't work, it seems like what you have here is probably sufficient.

Reply:
Done